app/coffee/DocManager.coffee

```diff
@@ -51,7 +52,13 @@ module.exports = DocManager =
 					newDocLines: lines
 					rev: doc.rev
 				}, "updating doc lines"
-				MongoManager.updateDoc project_id, mongoPath, lines, (error) ->
+				async.series [
+					(cb)->
+						# project collection is still primary so that needs to be successful first
+						MongoManager.updateDoc project_id, mongoPath, lines, cb
+					(cb)->
+						MongoManager.upsertIntoDocCollection project_id, doc_id, lines, doc.rev, cb
```

Comment:
It should be fine now as only the project collection is being read from. My plan was once we start reading from the doc collection first we flip these 2 functions around in order.